cmd/bisync/cmd.go

```diff
@@ -0,0 +1,224 @@
+}
+
+// Type of the CheckSync value
+func (x *CheckSyncMode) Type() string {
+	return "string"
+}
+
+// Opt keeps command line options
+var Opt Options
+
+func init() {
+	cmd.Root.AddCommand(commandDefinition)
+	cmdFlags := commandDefinition.Flags()
+	flags.BoolVarP(cmdFlags, &Opt.Resync, "resync", "1", Opt.Resync, "Performs the resync run. Path1 files may overwrite Path2 versions. Consider using --verbose or --dry-run first.")
+	flags.BoolVarP(cmdFlags, &Opt.CheckAccess, "check-access", "", Opt.CheckAccess, makeHelp("Ensure expected {CHECKFILE} files are found on both Path1 and Path2 filesystems, else abort."))
+	flags.StringVarP(cmdFlags, &Opt.CheckFilename, "check-filename", "", Opt.CheckFilename, makeHelp("Filename for --check-access (default: {CHECKFILE})"))
+	flags.BoolVarP(cmdFlags, &Opt.Force, "force", "", Opt.Force, "Bypass --max-delete safety check and run the sync. Consider using with --verbose")
+	flags.FVarP(cmdFlags, &Opt.CheckSync, "check-sync", "", "Controls comparison of final listings: true|false|only (default: true)")
+	flags.BoolVarP(cmdFlags, &Opt.RemoveEmptyDirs, "remove-empty-dirs", "", Opt.RemoveEmptyDirs, "Remove empty directories at the final cleanup step.")
+	flags.StringVarP(cmdFlags, &Opt.FiltersFile, "filters-file", "", Opt.FiltersFile, "Read filtering patterns from a file")
+	flags.StringVarP(cmdFlags, &Opt.Workdir, "workdir", "", Opt.Workdir, makeHelp("Use custom working dir - useful for testing. (default: {WORKDIR})"))
+	flags.BoolVarP(cmdFlags, &tzLocal, "localtime", "", tzLocal, "Use local time in listings (default: UTC)")
+	flags.BoolVarP(cmdFlags, &Opt.NoCleanup, "no-cleanup", "", Opt.NoCleanup, "Retain working files (useful for troubleshooting and testing).")
+}
+
+// bisync command definition
+var commandDefinition = &cobra.Command{
+	Use:   "bisync remote1:path1 remote2:path2",
+	Short: shortHelp,
+	Long:  longHelp,
+	RunE: func(command *cobra.Command, args []string) error {
+		cmd.CheckArgs(2, 2, command, args)
+		fs1, file1, fs2, file2 := cmd.NewFsSrcDstFiles(args)
+		if file1 != "" || file2 != "" {
+			return errors.New("paths must be existing directories")
+		}
+
+		ctx := context.Background()
+		opt := Opt
+		opt.applyContext(ctx)
+
+		if tzLocal {
+			TZ = time.Local
+		}
+
+		commonHashes := fs1.Hashes().Overlap(fs2.Hashes())
+		isDropbox1 := strings.HasPrefix(fs1.String(), "Dropbox")
+		isDropbox2 := strings.HasPrefix(fs2.String(), "Dropbox")
+		if commonHashes == hash.Set(0) && (isDropbox1 || isDropbox2) {
+			ci := fs.GetConfig(ctx)
+			if !ci.DryRun && !ci.RefreshTimes {
+				fs.Debugf(nil, "Using flag --refresh-times is recommended")
+			}
+		}
+
+		// FIXME Why runtime errors make rclone print usage?
+		command.SilenceUsage = true
+
+		err := Bisync(ctx, fs1, fs2, &opt)
+		if err == ErrBisyncAborted {
+			os.Exit(2)
+		}
+		return err
+	},
+}
+
+func (opt *Options) applyContext(ctx context.Context) {
+	maxDelete := DefaultMaxDelete
+	ci := fs.GetConfig(ctx)
+	if ci.MaxDelete >= 0 {
```

Comment:
@ncw 
I made bisync use the generic `--max-delete` flag but as a percentage in range `0..100` rather than absolute number of deleted objects.
Maybe it's better to rename it as `--max-delete-percent` or `--bisync-max-delete`?